File: prime-number/cmd.cpp
#include <stdio.h>
#include <stdlib.h>

int main( int argc, char *argv[] )
{
    int32_t primeNumberCount = 100;
    if (argc == 2) {
        primeNumberCount = atoi(argv[1]);
    }

    int32_t number = 0;
    int32_t primeNumberCountOrigin = primeNumberCount;
    while (primeNumberCount > 0) {
        ++number;

        int32_t j = 0;
        int32_t dividers[primeNumberCountOrigin];

        for (int32_t i = 1; i <= number; ++i) {
            if (number % i == 0) {
                dividers[j] = i;
                ++j;
            }
        }

        if (j == 2) {
            --primeNumberCount;
        }
    }

    printf("The latest prime number: %d\n", number);
}

File: prime-number/julia/cmd.jl
primeNumberCount = length(ARGS) == 1 ? parse(Int64, ARGS[1]) : 100

number = 0
while primeNumberCount > 0
    global number += 1
    j = 0

    for i in 1:number + 1
        if number % i == 0
            j += 1
        end
    end

    if j == 2
        global primeNumberCount -= 1
    end
end

println("The latest prime number: $number")
